src/Native/Runtime/windows/CoffNativeCodeManager.cpp

```diff
@@ -380,9 +380,44 @@ void CoffNativeCodeManager::EnumGcRefs(MethodInfo *    pMethodInfo,
 
 UIntNative CoffNativeCodeManager::GetConservativeUpperBoundForOutgoingArgs(MethodInfo * pMethodInfo, REGDISPLAY * pRegisterSet)
 {
-    // @TODO: CORERT: GetConservativeUpperBoundForOutgoingArgs
+#if defined(_TARGET_AMD64_)
+
+    // Return value
+    UIntNative upperBound;
+    CoffNativeMethodInfo* pNativeMethodInfo = (CoffNativeMethodInfo *) pMethodInfo;
+    
+    // Get the method's GC info
+    size_t unwindDataBlobSize;
+    PTR_VOID pUnwindDataBlob = GetUnwindDataBlob(m_moduleBase, pNativeMethodInfo->runtimeFunction, &unwindDataBlobSize);
+    PTR_UInt8 p = dac_cast<PTR_UInt8>(pUnwindDataBlob) + unwindDataBlobSize;
+    uint8_t unwindBlockFlags = *p++;
+    GcInfoDecoder decoder(GCInfoToken(p), DECODE_REVERSE_PINVOKE_VAR);
+    
+    UINT32 stackBasedRegister = decoder.GetStackBaseRegister();
+    // Initialize the base pointer to either  stack or frame pointer
+    TADDR basePointer = stackBasedRegister == NO_STACK_BASE_REGISTER ? dac_cast<TADDR>(pRegisterSet->GetSP()) : dac_cast<TADDR>(pRegisterSet->GetFP());
+    
+    if ((unwindBlockFlags & UBF_FUNC_REVERSE_PINVOKE) != 0)
+    {
+        // Reverse PInvoke case.  The embedded reverse PInvoke frame is guaranteed to reside above
+        // all outgoing arguments.
+        INT32 slot = decoder.GetReversePInvokeFrameStackSlot();
+        upperBound =  (UIntNative) dac_cast<TADDR>(basePointer + slot);
+    }
+    else
+    {
+        // In amd64, it is guaranteed that if there is a pushed RBP
```

Comment:
What happens if there is no RBP pushed? (`stackBasedRegister == NO_STACK_BASE_REGISTER` case above)

Reply:
Yes, this case is not handled here. @A-And, please take another look at the implementation in RhCodeMan. In addition, as we discussed a day ago, please add a few test cases to cover each branch in this code.

Reply:
In the case where no RBP is pushed, then basePointer will be set to RSP and the upper bound calculated below via the frame offset.